### PR TITLE
Adds Sync Functionality to Inventory Source List

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
this addresses https://github.com/ansible/awx/issues/6777

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```

```

##### ADDITIONAL INFORMATION
Sync

### CODE REVIEW

File: awx/ui_next/src/api/models/InventorySources.js [hunk 1]
```diff
@@ -5,7 +5,19 @@ class InventorySources extends LaunchUpdateMixin(Base) {
   constructor(http) {
     super(http);
     this.baseUrl = '/api/v2/inventory_sources/';
+
+    this.allowSyncStart = this.allowSyncStart.bind(this);
+    this.startSyncSource = this.startSyncSource.bind(this);
+  }
+
+  allowSyncStart(sourceId) {
```
Comment (jlmitch5) [inline awx/ui_next/src/api/models/InventorySources.js]: I'm not sure if we should name these based on what they are specifically doing, or if we should come up with something more based on the conventions we've been doing for the other api fn's.  Something like:

readSourceSyncUpdate
createSourceSyncUpdate

File: awx/ui_next/src/api/models/InventoryUpdates.js [hunk 2]
```diff
@@ -5,7 +5,16 @@ class InventoryUpdates extends LaunchUpdateMixin(Base) {
   constructor(http) {
     super(http);
     this.baseUrl = '/api/v2/inventory_updates/';
+    this.allowSyncCancel = this.allowSyncCancel.bind(this);
```
Comment (jlmitch5) [inline awx/ui_next/src/api/models/InventoryUpdates.js]: again, maybe

readSourceSyncCancel
createSourceSyncCancel

File: awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceListItem.jsx [hunk 3]
```diff
@@ -21,47 +22,67 @@ function InventorySourceListItem({
   detailUrl,
   label,
 }) {
+  const [isCancelSyncLoading, setIsCancelSyncLoading] = useState(false);
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceListItem.jsx]: I think we could just create a single isLoading state var...

File: awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceListItem.jsx [hunk 4]
```diff
@@ -21,47 +22,67 @@ function InventorySourceListItem({
...
+          />
+          <DataListAction
+            id="actions"
+            aria-labelledby="actions"
+            aria-label="actions"
+          >
+            {source.summary_fields.user_capabilities.start && (
+              <InventorySourceSyncButton
+                onCancelSyncLoading={isLoading =>
+                  setIsCancelSyncLoading(isLoading)
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceListItem.jsx]: ...which we set in both of these start and cancel functions

File: awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceSyncButton.jsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    useCallback(async () => {
+      let syncStatus;
+
+      const {
+        data: { can_update },
+      } = await InventorySourcesAPI.allowSyncStart(source.id);
+      if (can_update) {
+        syncStatus = await InventorySourcesAPI.startSyncSource(source.id);
+      } else {
+        throw new Error(
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceSyncButton.jsx]: do we need this can_update/can_cancel conditionals.  It looks like we already hide the button based on user_capabilities start boolean.  I'm fine with simplifying this component with the assumption it is only rendered if they have that permission based on that, unless there's some reason why further checking is needed.

File: awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceSyncButton.test.jsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    expect(InventoryUpdatesAPI.allowSyncCancel).toBeCalledWith(120);
+    expect(InventoryUpdatesAPI.cancelSyncSource).not.toBeCalledWith(120);
+
+    wrapper.update();
+    expect(wrapper.find('AlertModal').length).toBe(1);
+    expect(wrapper.find('Button[aria-label="Cancel sync source"]').length).toBe(
+      1
+    );
+  });
+});
```
Comment (jlmitch5) [inline awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceSyncButton.test.jsx:93]: great tests!

**Review by jlmitch5 — COMMENTED** [review]
I can quickly rereview and approve, just wanted to get your thoughts first on a few things

**Comment by dsesami** [comment]
Two questions:
1) Are there plans to include the sync status/result on this list? The current UI utilizes a red/green indicator with a cloud to identify inventories that have failed or succeeded their sync.
2) There is a `SYNC ALL` button available in the current UI. Are there plans to implement this as well?

File: awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceSyncButton.jsx (hunk 5 above)
Comment (AlexSCorey) [reply awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceSyncButton.jsx]: spoke with @jlmitch5 offline.  It will be removed.

**Comment by AlexSCorey** [comment]
> 2. There is a `SYNC ALL` button available in the current UI. Are there plans to implement this as well?

@dsesami We are going to have that button.  It probably is not within the scope of this PR.  That issue can be tracked here https://github.com/ansible/awx/issues/6903

**Review by dsesami — APPROVED** [review]
This looks good, and the status updates appropriately for sync. :+1:

**Review by jlmitch5 — APPROVED** [review]
looks good!

**Pull request merged**